src/shillelagh/filters.py

```diff
@@ -425,3 +426,36 @@ def __repr__(self) -> str:
             operator = "<=" if self.include_end else "<"
             comparisons.append(f"{operator}{self.end}")
         return ",".join(comparisons)
+
+
+class In(Filter):
+    """
+    Substring searches.
+    """
+
+    operators: Set[Operator] = {
+        Operator.IN,
+    }
+
+    def __init__(self, value: Any):
+        self.value = value
+
+    @classmethod
+    def build(cls, operations: Set[Tuple[Operator, Any]]) -> Filter:
+        # we only accept a single value
+        values = {value for operator, value in operations}
+        if len(values) != 1:
+            return Impossible()
+
+        # we only accept tuples
+        value = values.pop()
+        if not isinstance(value, tuple):
```

Comment:
I would probably be looser here with the types accepted.

Reply:
These come from https://rogerbinns.github.io/apsw/vtable.html#apsw.VTCursor.Filte. Only set is mentioned there I believe.

Reply:
I am not totally clear where this code gets called. I remember it being cases where the sql engine doesn't handle filtering.